### ISSUE TITLE
QueryComplexity does not work with ES Modules (ES2020, Typescript)

### ISSUE THREAD

**Issue by tvvignesh** [slicknode/graphql-query-complexity#37]
Hi. I have my codebase in Typescript and had `queryComplexity` setup in my project some time ago. It worked well when my codebase was transpiled to commonjs and I was targetting `ES2017`

But recently I made the move to use `ES2020` with Node.js modules (`type:"module" in `package.json`) and I get this error:

<img width="769" alt="e1" src="https://user-images.githubusercontent.com/1165845/100751689-a5741700-340d-11eb-9580-5638e3160f35.PNG">

This is how I am importing (as in the README):
`import queryComplexity, { directiveEstimator, simpleEstimator } from 'graphql-query-complexity';`

And my validationRules look like this (from the README):

```ts
validationRules: [
            queryComplexity({
              estimators: [
                directiveEstimator({
                  name: 'complexity'
                }),
                simpleEstimator({
                  defaultComplexity: 1
                })
              ],
              maximumComplexity: 500,
              variables: variables,
              onComplete: (complexity: number) => {
                console.log('Query Complexity:', complexity);
              },
              createError: (max: number, actual: number) => {
                return new GraphQLError(`Query is too complex: ${actual}. Maximum allowed complexity: ${max}`);
              }
            })
          ]
```

My old TSConfig (which worked):

```json
{
  "compileOnSave": true,
  "compilerOptions": {
    "target": "ES2017",
    "module": "commonjs",
    "declaration": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "moduleResolution": "node",
    "esModuleInterop": true,
    "incremental": true,
    "tsBuildInfoFile": "./dist/tsbuildinfo",
    "alwaysStrict": true,
    "skipLibCheck": true,
    "noImplicitAny": false,
    "noUnusedLocals": true,
    "removeComments": true,
    "preserveConstEnums": true,
    "allowSyntheticDefaultImports": true,
    "outDir": "dist",
    "rootDir": "src",
    "declarationMap": true,
    "sourceMap": true,
    "strict": true,
    "noEmitOnError": true,
    "lib": ["ES2017"]
  },
  "include": ["./src/**/*"],
  "exclude": ["node_modules", "**/*.spec.ts", "**/*.d.ts"]
}
```

My new TSConfig which does not work:

```json
{
	"compileOnSave": true,
	"compilerOptions": {
	  "target": "ES2020",
	  "module": "ESNext",
	  "declaration": true,
	  "experimentalDecorators": true,
	  "emitDecoratorMetadata": true,
	  "moduleResolution": "node",
	  "esModuleInterop": true,
	  "incremental": true,
	  "tsBuildInfoFile": "./dist/tsbuildinfo",
	  "alwaysStrict": true,
	  "skipLibCheck": true,
	  "noImplicitAny": false,
	  "noUnusedLocals": true,
	  "removeComments": true,
	  "preserveConstEnums": true,
	  "allowSyntheticDefaultImports": true,
	  "outDir": "dist",
	  "rootDir": "src",
	  "declarationMap": true,
	  "sourceMap": true,
	  "strict": true,
	  "noEmitOnError": true,
	  "lib": ["ES2020"]
	},
	"include": ["./src/**/*"],
	"exclude": ["node_modules", "**/*.spec.ts", "**/*.d.ts"]
  }
```

**PS:** All other libraries work except this since I have `esModuleInterop` set.

Any ideas what I might be doing wrong? Thanks.

**Comment by tvvignesh** [comment]
I tried working around like this, but even this did not work:

```ts
import { directiveEstimator, QueryComplexityOptions, simpleEstimator } from 'graphql-query-complexity';
import QueryComplexity from 'graphql-query-complexity/dist/QueryComplexity.js';

function queryComplexity(options: QueryComplexityOptions): (context: ValidationContext) => QueryComplexity {
  return (context: ValidationContext): QueryComplexity => {
    return new QueryComplexity(context, options);
  };
}
```

It said `QueryComplexity` is not a constructor.

**Comment by PythonCreator27** [comment]
Here is a workaround:
```ts
import gqlQueryComplexity from 'graphql-query-complexity';
type gqlQueryComplexityFn = typeof gqlQueryComplexity;
const queryComplexity = (
    gqlQueryComplexity as unknown as { default: gqlQueryComplexityFn }
).default;
```

**Comment by ivome** [comment]
This should be fixed with the latest release of v0.9.0. The package now includes an ESM build as well as a commonJS build. 
There is now also a named export for `createComplexityRule`. I updated the READMEs accordingly.
